ReactWindows/ReactNative/Touch/TouchHandler.cs

```diff
@@ -231,6 +232,10 @@ private void UpdatePointerForEvent(ReactPointer pointer, PointerPoint rootPoint,
             pointer.Timestamp = rootPoint.Timestamp / 1000; // Convert microseconds to milliseconds;
             pointer.Force = rootPoint.Properties.Pressure;
             pointer.IsBarrelButtonPressed = rootPoint.Properties.IsBarrelButtonPressed;
+
+            pointer.ShiftKey = Window.Current.CoreWindow.GetKeyState(Windows.System.VirtualKey.Shift).HasFlag(CoreVirtualKeyStates.Down);
+            pointer.AltKey = Window.Current.CoreWindow.GetKeyState(Windows.System.VirtualKey.Menu).HasFlag(CoreVirtualKeyStates.Down);
+            pointer.CtrlKey = Window.Current.CoreWindow.GetKeyState(Windows.System.VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down);
         }
```

Comment:
Do we want to implement this key or add an API for core window keyboard events to JavaScript? I guess the concern with doing it in JavaScript is that it's asynchronous (something could get out of sync between the keyboard state and the click event)

Reply:
Yes, I followed the "stamp the modifier keys at event creation moment" pattern. 
If by API you mean a notification pipe from native to JS that keeps some JS state in sync with the kbd state, I just didn't think the complexity would warrant the perf gain.


---
In reply to: [172177315](https://github.com/Microsoft/react-native-windows/pull/1693#discussion_r172177315) [](ancestors = 172177315)